modules/mobile/app/models/mobile/v2/adapters/appointments.rb

```diff
@@ -0,0 +1,328 @@
+            APPOINTMENT_TYPES[:va_video_connect_home],
+            APPOINTMENT_TYPES[:va_video_connect_gfe]
+            if telehealth
+              address = telehealth.dig(:atlas, :address)
+
+              if address
+                location[:address] = {
+                  street: address[:street_address],
+                  city: address[:city],
+                  state: address[:state],
+                  zip_code: address[:zip_code],
+                  country: address[:country]
+                }
+              end
+
+              location[:url] = telehealth[:url]
+              location[:code] = telehealth.dig(:atlas, :confirmation_code)
+            end
+          else
+            location[:id] = appointment_hash.dig(:location, :id)
+            location[:name] = appointment_hash.dig(:location, :name)
+            address = appointment_hash.dig(:location, :physical_address)
+            if address.present?
+              location[:address] = {
+                street: address[:line]&.join(' ')&.strip,
+                city: address[:city],
+                state: address[:state],
+                zip_code: address[:postal_code]
+              }
+            end
+            location[:lat] = appointment_hash.dig(:location, :lat)
+            location[:long] = appointment_hash.dig(:location, :long)
+            location[:phone] = location_phone(appointment_hash)
+          end
+
+          location
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        def location_phone(appointment_hash)
+          phone = appointment_hash.dig(:location, :phone, :main)
+          return nil unless phone
+
+          # captures area code (\d{3}) number (\d{3}-\d{4})
+          # and optional extension (until the end of the string) (?:\sx(\d*))?$
+          phone_captures = phone.match(/^(\d{3})-(\d{3}-\d{4})(?:\sx(\d*))?$/)
+
+          if phone_captures.nil?
+            Rails.logger.warn(
+              'mobile appointments failed to parse VAOS V2 facility phone number',
+              facility_id: appointment_hash.dig(:location, :id),
+              facility_phone: phone
+            )
+            return nil
+          end
+
+          {
+            area_code: phone_captures[1].presence,
+            number: phone_captures[2].presence,
+            extension: phone_captures[3].presence
+          }
+        end
+
+        def time_zone(facility_id)
+          facility = Mobile::VA_FACILITIES_BY_ID["dfn-#{facility_id}"]
+          facility ? facility[:time_zone] : nil
+        end
+
+        def healthcare_provider(practitioners)
+          return nil if Array.wrap(practitioners&.map { |prac| prac[:name] })&.compact == []
+
+          practitioners_names = practitioners.map do |practitioner|
+            name = "#{practitioner&.dig(:name, :given)&.join(' ')&.strip}, #{practitioner&.dig(:name, :family)}"
+            '' if name == ', '
+            name
+          end
+          practitioners_names.join("\n").strip
```

Comment:
assuming front end could handle, new lines. need to verify

Reply:
What's the reason that we need to use new lines here?

Reply:
I couldn't think of a natural, human readable way to put multiple practitioners full names on the same line in a human readable way while I was already using commas to separate the given and last name. Though thinking about it again, I guess there isn't a reason to separate the given name with the last name with a comma. I think I was just afraid of what could be contained in the `given` array but it's most likely just the first name and sometimes the middle name.